### PR TITLE
Feature/additional reactor debug

### PR DESCRIPTION
**Pull request by n-hutton**
This PR fixes a bug in the HTTP server when a connection was closing - effectively a socket could get close called on it twice. It also fixes a backward compatability bug in the stake manager, a potential deadlock in promise, and makes some performance improvements to the chain sync, centering around bloom filtering.

### CODE REVIEW

**Review by ejfitzgerald — DISMISSED** [review]

File: libs/ledger/include/ledger/consensus/stake_manager.hpp [hunk 1]
```diff
@@ -167,7 +167,7 @@ struct MapSerializer<ledger::StakeManager, D>
   template <typename Constructor>
   static void Serialize(Constructor &map_constructor, Type const &stake_manager)
   {
-    auto map = map_constructor(4);
+    auto map = map_constructor(5);
```
Comment (n-hutton) [inline libs/ledger/include/ledger/consensus/stake_manager.hpp:170]: bugfix is here

File: libs/ledger/src/chain/main_chain.cpp [hunk 2]
```diff
@@ -1477,6 +1478,17 @@ BlockStatus MainChain::InsertBlock(IntBlockPtr const &block, bool evaluate_loose
     }
   }
 
+  if (evaluate_loose_blocks)  // normal case - not being called from inside CompleteLooseBlocks
```
Comment (n-hutton) [inline libs/ledger/src/chain/main_chain.cpp:1481]: this is a performance improvement that avoids detecting duplicate TXs unneccessarily

File: libs/ledger/src/chain/main_chain.cpp [hunk 3]
```diff
@@ -2001,7 +2006,10 @@ DigestSet MainChain::DetectDuplicateTransactions(BlockHash const &           sta
   DigestSet potential_duplicates{};
   for (auto const &tx_layout : transactions)
   {
-    if (bloom_filter_.Match(tx_layout.digest(), tx_layout.valid_from(), tx_layout.valid_until()))
+    auto const default_valid_from =
```
Comment (n-hutton) [inline libs/ledger/src/chain/main_chain.cpp:2009]: be slightly less aggressive with bloom searches

File: libs/ledger/src/protocols/main_chain_rpc_service.cpp [hunk 4]
```diff
@@ -237,7 +254,7 @@ MainChainRpcService::Address MainChainRpcService::GetRandomTrustedPeer() const
 
   auto const direct_peers = endpoint_.GetDirectlyConnectedPeers();
 
-  FETCH_LOG_DEBUG(LOGGING_NAME, "Main chain connected peers: ", direct_peers.size());
+  FETCH_LOG_INFO(LOGGING_NAME, "Main chain connected peers: ", direct_peers.size());
```
Comment (n-hutton) [inline libs/ledger/src/protocols/main_chain_rpc_service.cpp]:
```suggestion
  FETCH_LOG_DEBUG(LOGGING_NAME, "Main chain connected peers: ", direct_peers.size());
```

**Pull request merged**